xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/CoinmateStreamingService.java

```diff
@@ -0,0 +1,53 @@
+package info.bitrich.xchangestream.coinmate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinmate.dto.auth.AuthParams;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import sun.rmi.runtime.Log;
+
+import java.io.IOException;
+
+/**
+ * The new API uses one WebSocket connection per channel, unlike other exchanges,
```

Comment:
This could also be achieved be expecting the `ProductSubscription`s to be passed on `connect()`, in the same way as `BinanceStreamingExchange`.  I _think_ this is better, since it removes the need to disconnect/reconnect to change subscriptions - something that always confuses Binance users - but we should be aware we're doing it (and probably move Binance to this approach instead and get rid of `ProductSubscription`)

Something to think about.